MetaMask/metamask-extension
# Deprecated Icon and ButtonIcon clean up

## Explanation
Currently there are still some instances of the deprecated icon and button icon components as well as constants `ICON_NAMES` and `BUTTON_ICON_SIZES` in the codebase. This PR removes the instances and scripts associated with the deprecated components

* Fixes #18704 
* Fixes #18493

## Screenshots/Screencaps
Going to add the individual screenshots as code comments for better context and easier review. If you don't see them scroll down

## Manual Testing Steps
Manual testing for this would be quite laborious so I have gone to the effort of taking before/after screenshots of all instances of the components that I am replacing in the code comments. But if you would like to do some additional manual checks you could
- Pull this branch
- Search for instances of `ICON_NAMES`, `BUTTON_ICON_SIZES`, `icon/deprecated`, `generate-icon-names`, `generateIconNames`
- Make sure there are no instances outside of the CHANGELOG

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
